catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -155,7 +164,9 @@ public void createContentItem(FileSystemPersistenceProvider fileIdMap, ContentEn
                 List<Metacard> createdMetacards = createResponse.getCreatedMetacards();
 
                 for (Metacard metacard : createdMetacards) {
-                    fileIdMap.store(key, metacard.getId());
+                    if (safeKey != null) {
```

Comment:
I'm okay either way on it. I just thought it looked a bit odd. If you think there is value in logging every content item created with its content ID I would leave it. Otherwise I would refactor away because it could be a lot of noise and not as valuable to an admin looking at the log.